pkg/sql/scrub_physical.go

```diff
@@ -145,11 +145,12 @@ func (o *physicalCheckOperation) Start(params runParams) error {
 	o.primaryColIdxs = primaryColIdxs
 	o.columns = columns
 	o.run.started = true
-	o.run.rows, err = scrubRunDistSQL(ctx, &planCtx, params.p, &physPlan, distsqlrun.ScrubTypes)
+	rows, err := scrubRunDistSQL(ctx, &planCtx, params.p, &physPlan, distsqlrun.ScrubTypes)
 	if err != nil {
-		o.run.rows.Close(ctx)
+		rows.Close(ctx)
```

Comment:
I don't understand why this will help - this doesn't seem different from the old code path. You're calling Close() on the results of scrubRunDistSQL in both cases right?

Reply:
I don't understand the change either. Vivek can you explain a bit more?

Reply:
`scrubNode.Close()` calls `physicalCheckOperation.Close()` on every `physicalCheckOperation`, and `physicalCheckOperation.Close()` in turn calls `o.run.rows.Close()`. This caused a double close: once, here, in the error branch, and once when `scrubNode.Close()` was called.

The solution here is to avoid assigning `o.run.rows` in the error branch. (I think not calling `rows.Close()` in the error branch would work equally well, except that it might violate the usual semantics of `Close()`—i.e., that it need not be called when an error is returned?)

Reply:
Oh right. Thank you!